// tokopedia-gql/dist/api/search/constants.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.SHIPPING = exports.CONDITION = exports.SEARCH_TYPE = exports.ORDER_BY = exports.QUERY = exports.PARAMETER_DELIMITER = exports.NORMALIZED_SEARCH_PARAMETER_KEY = exports.ENDPOINT = void 0;
exports.ENDPOINT = "https://gql.tokopedia.com/";
exports.NORMALIZED_SEARCH_PARAMETER_KEY = {
    orderBy: "ob",
    goldMerchant: "goldmerchant",
    query: "q",
    rating: "rt",
    priceMin: "pmin",
    priceMax: "pmax",
    searchType: "st",
    safeSearch: "safe_search"
};
exports.PARAMETER_DELIMITER = {
    condition: "#",
    shipping: "#",
    rt: ","
};
exports.QUERY = "\nquery SearchProductQueryV4($params: String!) {\n\tace_search_product_v4(params: $params) {\n\t\theader {\n\t\ttotalData\n\t\ttotalDataText\n\t\tprocessTime\n\t\tresponseCode\n\t\terrorMessage\n\t\tadditionalParams\n\t\tkeywordProcess\n\t\t__typename\n\t\t}\n\t\tdata {\n\t\tisQuerySafe\n\t\tticker {\n\t\t\ttext\n\t\t\tquery\n\t\t\ttypeId\n\t\t\t__typename\n\t\t}\n\t\tredirection {\n\t\t\tredirectUrl\n\t\t\tdepartmentId\n\t\t\t__typename\n\t\t}\n\t\trelated {\n\t\t\trelatedKeyword\n\t\t\totherRelated {\n\t\t\tkeyword\n\t\t\turl\n\t\t\tproduct {\n\t\t\t\tid\n\t\t\t\tname\n\t\t\t\tprice\n\t\t\t\timageUrl\n\t\t\t\trating\n\t\t\t\tcountReview\n\t\t\t\turl\n\t\t\t\tpriceStr\n\t\t\t\twishlist\n\t\t\t\tshop {\n\t\t\t\tcity\n\t\t\t\tisOfficial\n\t\t\t\tisPowerBadge\n\t\t\t\t__typename\n\t\t\t\t}\n\t\t\t\tads {\n\t\t\t\tid\n\t\t\t\tproductClickUrl\n\t\t\t\tproductWishlistUrl\n\t\t\t\tshopClickUrl\n\t\t\t\tproductViewUrl\n\t\t\t\t__typename\n\t\t\t\t}\n\t\t\t\t__typename\n\t\t\t}\n\t\t\t__typename\n\t\t\t}\n\t\t\t__typename\n\t\t}\n\t\tsuggestion {\n\t\t\tcurrentKeyword\n\t\t\tsuggestion\n\t\t\tsuggestionCount\n\t\t\tinstead\n\t\t\tinsteadCount\n\t\t\tquery\n\t\t\ttext\n\t\t\t__typename\n\t\t}\n\t\tproducts {\n\t\t\tid\n\t\t\tname\n\t\t\tads {\n\t\t\tid\n\t\t\tproductClickUrl\n\t\t\tproductWishlistUrl\n\t\t\tproductViewUrl\n\t\t\t__typename\n\t\t\t}\n\t\t\tbadges {\n\t\t\ttitle\n\t\t\timageUrl\n\t\t\tshow\n\t\t\t__typename\n\t\t\t}\n\t\t\tcategory: departmentId\n\t\t\tcategoryBreadcrumb\n\t\t\tcategoryId\n\t\t\tcategoryName\n\t\t\tcountReview\n\t\t\tdiscountPercentage\n\t\t\tgaKey\n\t\t\timageUrl\n\t\t\tlabelGroups {\n\t\t\tposition\n\t\t\ttitle\n\t\t\ttype\n\t\t\t__typename\n\t\t\t}\n\t\t\toriginalPrice\n\t\t\tprice\n\t\t\tpriceRange\n\t\t\trating\n\t\t\tshop {\n\t\t\tid\n\t\t\tname\n\t\t\turl\n\t\t\tcity\n\t\t\tisOfficial\n\t\t\tisPowerBadge\n\t\t\t__typename\n\t\t\t}\n\t\t\turl\n\t\t\twishlist\n\t\t\tsourceEngine: source_engine\n\t\t\t__typename\n\t\t}\n\t\t__typename\n\t\t}\n\t\t__typename\n\t}\n}";
exports.ORDER_BY = {
    LOWEST_PRICE: 3,
    HIGHEST_PRICE: 4,
    REVIEW: 5,
    NEWEST: 9,
    MOST_RELEVANT: 23
};
exports.SEARCH_TYPE = {
    PRODUCT: "product",
    SHOP: "shop"
};
exports.CONDITION = {
    NEW: 1,
    USED: 2
};
exports.SHIPPING = {
    JNE: 1,
    TIKI: 2,
    POS_INDONESIA: 4,
    WAHANA: 6,
    GO_SEND: 10,
    SICEPAT: 11,
    NINJA_EXPRESS: 12,
    GRAB_EXPRESS: 13,
    JNT: 14,
    REX: 16,
    ANTERAJA: 23,
    LION_PARCEL: 24
};
